shiny/ui/_layout_columns.py

```diff
@@ -0,0 +1,305 @@
+          will repeat. For example, `row_heights=["auto", 1]` allows the height of odd
+          rows to be driven my it's contents and even rows to be
+          [`1fr`](https://css-tricks.com/introduction-fr-css-unit/).
+        * A single string containing CSS length units. In this case, the value is
+          supplied directly to `grid-auto-rows`.
+        * A dictionary of row heights at different breakpoints, where each key is a
+          breakpoint name (one of `"xs"`, `"sm"`, `"md"`, `"lg"`, `"xl"`, or `"xxl"`)
+          and where the values may be any of the above options.
+
+    fill
+        Whether or not to allow the layout to grow/shrink to fit a fillable container
+        with an opinionated height (e.g., :func:`~shiny.ui.page_fillable`).
+
+    fillable
+        Whether or not each element is wrapped in a fillable container.
+
+    gap
+        Any valid CSS unit to use for the gap between columns.
+
+    class_
+        CSS class(es) to apply to the containing element.
+
+    height
+        Any valid CSS unit to use for the height.
+
+    **kwargs
+        Additional attributes to apply to the containing element.
+
+    Returns
+    -------
+    :
+        An :class:`~htmltools.Tag` element.
+
+    See Also
+    --------
+    * :func:`~shiny.ui.layout_column_wrap` for laying out elements into a uniform grid.
+
+    Reference
+    --------
+    * [Bootstrap CSS Grid](https://getbootstrap.com/docs/5.3/layout/grid/)
+    * [Bootstrap Breakpoints](https://getbootstrap.com/docs/5.3/layout/breakpoints/)
+    """
+    attrs, children = consolidate_attrs(*args, class_=class_, **kwargs)
+
+    col_widths_spec = as_col_spec(col_widths, len(children))
+
+    # Create the bslib-layout-columns element
+    tag = Tag(
+        "bslib-layout-columns",
+        {
+            "class": "bslib-grid grid",
+            "style": css(
+                gap=as_css_unit(gap),
+                height=as_css_unit(height),
+            ),
+        },
+        col_widths_attrs(col_widths_spec),
+        attrs,
+        row_heights_attrs(row_heights),
+        *wrap_all_in_grid_item_container(children, fillable),
+        web_component_dependency(),
+    )
+
+    # Apply fill and fillable
+    if fill:
+        tag = as_fill_item(tag)
+    if fillable:
+        tag = as_fillable_container(tag)
+
+    return tag
+
+
+def wrap_all_in_grid_item_container(
+    children: Iterable[TagChild],
+    fillable: bool = True,
+    class_: Optional[str] = None,
+) -> list[TagChild]:
+    item_class = "bslib-grid-item bslib-gap-spacing"
+    if class_ is not None:
+        item_class = f"{item_class} {class_}"
+
+    # Use a new list so that we don't mutate the original `children`
+    wrapped_children: list[TagChild] = []
+    for child_value in children:
+        child = div({"class": item_class}, child_value)
+        if fillable:
+            child = as_fillable_container(child)
+        wrapped_children.append(child)
+
+    return wrapped_children
```

Comment:
`layout_column_wrap()` should use this too?

Reply:
Probably! This isn't a drop-in replacement for what happens to the children in  `layout_column_wrap()`, so I'd like to do this in a follow-up PR.

Reply:
👉 https://github.com/posit-dev/py-shiny/issues/899